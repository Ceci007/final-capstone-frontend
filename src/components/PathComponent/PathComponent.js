import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const PathComponent = ({
  path, icon, linkText, handleClick,
}) => (
  <Link to={`/${path}`} onClick={e => handleClick && handleClick(e)}>
    <div className="icons">

      <i className={`fa ${icon}`} />
      <p>{linkText}</p>
    </div>

  </Link>
);

PathComponent.propTypes = {
  path: PropTypes.string.isRequired,
  icon: PropTypes.string,
  linkText: PropTypes.string,
  handleClick: PropTypes.func,
};

PathComponent.defaultProps = {
  icon: '',
  linkText: '',
  handleClick: () => {},

};
export default PathComponent;
