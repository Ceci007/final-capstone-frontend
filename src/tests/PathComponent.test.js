import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import PathComponent from '../components/PathComponent/PathComponent';

test('PathComponent renders correctly', () => {
  const tree = create(
    <Router>
      <PathComponent />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
