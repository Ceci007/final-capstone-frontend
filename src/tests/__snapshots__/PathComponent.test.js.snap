// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<a
  href="/undefined"
  onClick={[Function]}
>
  <div
    className="icons"
  >
    <i
      className="fa "
    />
    <p>
      
    </p>
  </div>
</a>
`;

exports[`PathComponent renders correctly 1`] = `
<a
  href="/undefined"
  onClick={[Function]}
>
  <div
    className="icons"
  >
    <i
      className="fa "
    />
    <p>
      
    </p>
  </div>
</a>
`;
